blocks/embed/embed.js

```diff
@@ -0,0 +1,124 @@
+      scrolling="no" allow="encrypted-media" title="Content from ${url.hostname}" loading="lazy">
+    </iframe>
+  </div>`;
+
+const embedYoutube = (url) => {
+  const usp = new URLSearchParams(url.search);
+  let vid = usp.get('v');
+  const embed = url.pathname;
+  if (url.origin.includes('youtu.be')) {
+    vid = url.pathname.substring(1);
+  }
+  const embedHTML = `<div style="left: 0; width: 100%; height: 0; position: relative; padding-bottom: 56.25%;">
+      <iframe src="https://www.youtube.com${vid ? `/embed/${vid}?rel=0&amp;v=${vid}` : embed}" style="border: 0; top: 0; left: 0; width: 100%; height: 100%; position: absolute;" allow="encrypted-media; accelerometer; gyroscope; picture-in-picture" allowfullscreen="" scrolling="no" title="Content from Youtube" loading="lazy"></iframe>
+    </div>`;
+  return embedHTML;
+};
+
+const embedInstagram = (url) => {
+  const endingSlash = url.pathname.endsWith('/') ? '' : '/';
+  const location = window.location.href.endsWith('.html') ? window.location.href : `${window.location.href}.html`;
+  const src = `${url.origin}${url.pathname}${endingSlash}embed/captioned/?cr=1&amp;v=10&amp;wp=1080&amp;rd=${location}`;
+  const embedHTML = `<div>
+    <iframe class="instagram-media instagram-media-rendered" id="instagram-embed-0" src="${src}" allowtransparency="true" allowfullscreen="true" frameborder="0" height="975" data-instgrm-payload-id="instagram-media-payload-0" scrolling="no" style="background: white; max-width: 540px; width: calc(100% - 2px); border-radius: 3px; border: 1px solid rgb(219, 219, 219); box-shadow: none; display: block; margin: 0px 0px 12px; min-width: 326px; padding: 0px;"></iframe>
+    </div>`;
+  return embedHTML;
+};
+
+const embedVimeo = (url) => {
+  const video = url.pathname.split('/')[1];
+  const embedHTML = `<div style="left: 0; width: 100%; height: 0; position: relative; padding-bottom: 56.25%;">
+      <iframe src="https://player.vimeo.com/video/${video}" 
+      style="border: 0; top: 0; left: 0; width: 100%; height: 100%; position: absolute;" 
+      frameborder="0" allow="autoplay; fullscreen; picture-in-picture" allowfullscreen  
+      title="Content from Vimeo" loading="lazy"></iframe>
+    </div>`;
+  return embedHTML;
+};
+
+const embedTwitter = (url) => {
+  const embedHTML = `<blockquote class="twitter-tweet"><a href="${url}"></a></blockquote>`;
+  loadScript('https://platform.twitter.com/widgets.js');
+  return embedHTML;
+};
+
+const embedTiktok = (url) => {
+  const resultHtml = document.createElement('div');
+  resultHtml.setAttribute('id', 'tiktok');
+
+  const tiktokBuild = async (fetchUrl) => {
+    loadScript('https://www.tiktok.com/embed.js');
+    const response = await fetch(fetchUrl);
+    const json = await response.json();
+    const tiktok = document.getElementById('tiktok');
+    tiktok.outerHTML = json.html;
+  };
+  tiktokBuild(`https://www.tiktok.com/oembed?url=${url}`);
```

Comment:
Adding the `await` (and the corresponding `async` chain) should certainly fix the intermittent failures...

```suggestion
  await tiktokBuild(`https://www.tiktok.com/oembed?url=${url}`);
```